telegram/telegram.py

```diff
@@ -0,0 +1,199 @@
+import json
+
+from treq.client import HTTPClient
+
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.web import http
+from twisted.web.client import Agent
+from twisted.web._newclient import ResponseFailed
+
+from vumi.transports.httprpc.httprpc import HttpRpcTransport, HttpRpcResource
+from vumi.config import ConfigText
+from vumi import log
+
+
+class TelegramTransportConfig(HttpRpcTransport.CONFIG_CLASS):
+    bot_username = ConfigText(
+        'The username of our Telegram bot',
+        static=True,
+        required=True,
+    )
+    bot_token = ConfigText(
+        "Our bot's unique token to access the Telegram API",
+        static=True,
+        required=True,
+    )
+    outbound_url = ConfigText(
+        'The URL our bot should make requests to',
+        default='https://api.telegram.org/bot', static=True,
+    )
+    inbound_url = ConfigText(
+        'The URL our transport will listen on for Telegram updates',
+        static=True, required=True,
+    )
+
+
+class TelegramTransport(HttpRpcTransport):
+    """
+    Telegram transport for Vumi
+    """
+    transport_type = 'telegram'
+    transport_name = 'telegram_transport'
+    _requests = {}
```

Comment:
This doesn't seem to be used by anything and if it is, I think the underlyin `HttpRpcTransport` handles most of that request juggling for you.